spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -19,27 +19,56 @@ def foobar_action
 
 end
 
-RSpec.describe Hq::PartnersController, type: :controller do
+RSpec.describe Hq::PartnersController, type: :controller, focus: true do
+
   before :each do
     sign_in instance_double(AdminUser)
-    @partner = build_stubbed :partner
-    expect(Partner).to receive(:find).and_return(@partner)
   end
 
-  specify 'editing renders the edit view' do
-    get :edit, id: @partner.id
-    expect(response).to render_template 'edit'
-  end
+  context '#new and #create' do
+    let(:partner_double) { instance_double Partner, :attributes= => true }
+    
+    before :each do
+      controller.instance_variable_set(:@partner, partner_double)
+    end
+    
+    specify 'successful create redirects to the show view' do
+      expect(partner_double).to receive(:save).and_return(true)
+      post :create, partner: {}
+      expect(response).to redirect_to hq_partner_path(partner_double)
+    end
+
+    specify 'unsuccessful create renders the new view' do
+      expect(partner_double).to receive(:save).and_return(false)
+      post :create, partner: {}
+      expect(response).to render_template 'new'
+    end
 
-  specify 'unsuccessful update renders the edit view' do
-    expect(@partner).to receive(:save).and_return(false)
-    patch :update, id: @partner.id
-    expect(response).to render_template 'edit'
   end
 
-  specify 'successful update redirects to the show view' do
-    expect(@partner).to receive(:save).and_return(true)
-    patch :update, id: @partner.id
-    expect(response).to redirect_to(hq_partner_path(@partner))
+  context '#edit and #update' do 
```

Comment:
I really like the way `#edit & #update` have been implemented, by routing the instance variable setting through a stubbed model call instead of setting it directly